l1-contracts/contracts/common/libraries/FullMerkle.sol

```diff
@@ -143,4 +143,25 @@ library FullMerkle {
     function root(FullTree storage self) internal view returns (bytes32) {
         return self._nodes[self._height][0];
     }
+
+    /**
+     * @dev Returns merkle path for a certain leaf index.
+     * @param _index The index of the leaf to calculate proof for.
+     */
+    function merklePath(FullTree storage self, uint256 _index) public view returns (bytes32[] memory) {
+        // solhint-disable-next-line gas-custom-errors
```

Comment:
we need to use custom errors everywhere (sorry if not yet the full code adhere to it, but generally all should and it is fixed in a separate pr)